### ISSUE TITLE
Documenting up what we have under standalone web service

### ISSUE THREAD

**Issue by jarohen** [xtdb/xtdb#479]
* Can we break this down into various modules?
* What's the quality of those modules? (stable, beta, alpha)
* Where do those modules _really_ belong?

**Comment by danmason** [comment]
Currently, the webservice starts up:
- Crux HTTP Node/Server.
- A full webpage making use of this node.
- A REPL

Furthermore, there is the following in the folder (mostly for deployment):
- Kubernetes
- AWS 
- Backup and restore functionality

I would argue some of the extras are somewhat better suited to be in their own examples/modules (ie: the website makes use of crux.backup functionality to back up the contents every hour, which may be better to have as it's own example and arguably serves to over complicate the example).

**Comment by danmason** [comment]
I believe the primary example should be a service for the website connected to a Crux node, as it serves as a good example of bitemporality. Everything else included could be served as a separate example (as backup and restore already is).

Website and the node seem relatively *stable*, need to go through backup and restore some more to figure where it stands.

**Comment by jarohen** [comment]
Thanks @danmason :smile: as discussed at standup this morning, we're not going to be maintaining this sample webservice in the long term - we're looking to split it out into the various valuable parts and move them to new homes under the current Crux structure.

Particularly:
* Kubernetes to be included under #385, likely under the crux-http docker image that you've been working on recently
* AWS and backup, likewise, belong under #385 - using the rule-of-thumb that if it's something Crux should be providing, we move it; if it's not, we document it up for users to make their own decisions.
* The web UI should be included in discussions about the future of the console

Could you write up a little more on here about what this service makes possible for users for AWS, Kubernetes, and backup/restore?

**Comment by danmason** [comment]
- The webservice first calls `check-and-restore` from `crux.backup`, if there is not currently a db-dir/event-log-dir present it calls `bin/restore.sh`, which attempts to copy a **backup.tar.gz** file from **s3** via the **AWS CLI**. If there is a  **backup.tar.gz** file present after this, the file is untarred within a checkpoint directory, and the db-dir and event-log-dir are copied and the checkpoint directory deleted.
- Once an hour, the webservice calls `backup-current-version`, a checkpoint folder is made, a checkpoint is written within the folder, and it calls `bin/bin.sh` which tars the checkpoint directory, copies it to **s3** via the **AWS CLI**, and then the checkpoint directory is deleted.

TLDR: With regards to AWS & Backup, this service makes it possible for users to make hourly backups of their crux nodes which are saved in s3.

**Comment by jarohen** [comment]
ah, I see - so there's nothing about deploying to AWS?

**Comment by danmason** [comment]
I don't think so - the `deployment.sh` script seems to be for creating kubernetes containers.

**Comment by jarohen** [comment]
aaah. :bulb:

**Comment by danmason** [comment]
Any further thoughts on this, @jarohen?

**Comment by jarohen** [comment]
all good, cheers @danmason
